ui/src/dashboards/components/DashboardImportFromTemplateOverlay.tsx

```diff
@@ -0,0 +1,230 @@
+// Libraries
+import React, {PureComponent} from 'react'
+import {withRouter, WithRouterProps} from 'react-router'
+import {connect} from 'react-redux'
+import _ from 'lodash'
+
+// Components
+import {Button, ComponentColor} from '@influxdata/clockface'
+import {
+  Overlay,
+  ResponsiveGridSizer,
+  Panel,
+  EmptyState,
+  ComponentSize,
+} from 'src/clockface'
+import {
+  TemplateSummary,
+  ITemplate,
+  Organization,
+  TemplateType,
+  IDashboardTemplateIncluded,
+} from '@influxdata/influx'
+import CardSelectCard from 'src/clockface/components/card_select/CardSelectCard'
+import OrgTemplateFetcher from 'src/organizations/components/OrgTemplateFetcher'
+
+// Actions
+import {createDashboardFromTemplate as createDashboardFromTemplateAction} from 'src/dashboards/actions'
+import {getTemplateByID} from 'src/templates/actions'
+
+// Types
+import {AppState, RemoteDataState, DashboardTemplate} from 'src/types'
+
+interface StateProps {
+  templates: TemplateSummary[]
+  templateStatus: RemoteDataState
+  orgs: Organization[]
+}
+
+interface DispatchProps {
+  createDashboardFromTemplate: typeof createDashboardFromTemplateAction
+}
+
+interface State {
+  selectedTemplateSummary: TemplateSummary
+  selectedTemplate: ITemplate
+  variables: string[]
+  cells: string[]
+}
+
+class DashboardImportFromTemplateOverlay extends PureComponent<
+  StateProps & WithRouterProps & DispatchProps,
+  State
+> {
+  constructor(props) {
+    super(props)
+    this.state = {
+      selectedTemplateSummary: null,
+      selectedTemplate: null,
+      variables: [],
+      cells: [],
+    }
+  }
+
+  render() {
+    const {selectedTemplateSummary} = this.state
+
+    return (
+      <Overlay visible={true}>
+        <Overlay.Container maxWidth={800}>
+          <Overlay.Heading
+            title={`Import Dashboard from a Template`}
+            onDismiss={this.onDismiss}
+          />
+          <Overlay.Body>
+            <div className="import-template-overlay">
+              <OrgTemplateFetcher orgName={this.orgName}>
+                <ResponsiveGridSizer columns={3}>
+                  {this.templates}
+                </ResponsiveGridSizer>
+              </OrgTemplateFetcher>
+              {!selectedTemplateSummary && (
+                <Panel className="import-template-overlay--empty">
```

Comment:
Could we extract the empty state? There is a lot going on in this render function.